keeper/valuation_engine_test.go

```diff
@@ -0,0 +1,233 @@
+package keeper_test
+
+import (
+	"cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	markertypes "github.com/provenance-io/provenance/x/marker/types"
+
+	"github.com/provlabs/vault/keeper"
+	"github.com/provlabs/vault/types"
+	"github.com/provlabs/vault/utils"
+)
+
+func (s *TestSuite) TestUnitPriceFraction_Table() {
```

Comment:
Is there anyway to test that we won't get something back with a 0 in the denominator?

Reply:
Kind of hard to test.  The marker validates that the volume is not zero.  Which is how I could force a division of zero test.  I do check that volume is not zero before doing division: ```volAmt := math.NewInt(int64(nav.Volume))
	if volAmt.IsZero() {
		return math.Int{}, math.Int{}, fmt.Errorf("nav volume is zero for %s/%s", srcDenom, underlyingAsset)
	}``` ...